pkg/integrations/manager.go

```diff
@@ -224,25 +227,45 @@ func (m *Manager) runIntegration(ctx context.Context, cfg Config, i Integration)
 	}
 }
 
-func (m *Manager) instanceConfigForIntegration(cfg Config, i Integration) instance.Config {
+func (m *Manager) instanceConfigForIntegration(cfg Config, serverConfig serverconfig.ServerConfig, i Integration) instance.Config {
 	prometheusName := fmt.Sprintf("integration/%s", cfg.Name())
 
 	common := cfg.CommonConfig()
 	relabelConfigs := append(m.defaultRelabelConfigs, common.RelabelConfigs...)
 
 	var scrapeConfigs []*config.ScrapeConfig
+	schema := "http"
+
+	// If there is a TLS cert path then assume we need to use client certificates
+	httpClientConfig := config_util.HTTPClientConfig{}
+	if serverConfig.HTTPTLSConfig.TLSCertPath != "" {
+		if serverConfig.ClientKey == "" || serverConfig.ClientCert == "" {
+			err := errors.New("client key or client cert was not specific but TLS Cert was")
+			level.Error(m.logger).Log("tls", "Client TLS", "err", err)
+			panic(err)
```

Comment:
The panic here scares me a little bit. Let's check for this when the integrations manager is created and return an error if it's not set so we can avoid this.

Reply:
Cool will change, the main thing is if it requires a client key I didnt want it to keep running and just erroring out.